### PR TITLE
[Fix #1571] - Do not write to ActiveModel::Dirty changes when assigning ...

### LINKED ISSUE #1571: ActiveRecord "column"_changed? method issue with nil values
I'm using CarrierWave backgrounder which relies on `xxx_changed?` method to enqueue processing jobs. In a failing test I discovered this CW behavior which triggers background jobs when files are not present at all.

Given this class

```ruby
class Source < ActiveRecord::Base
  # CarrierWave integration for logos
  mount_uploader :logo, LogoUploader
end
```

the following happens

```
[12] pry(main)> s = Source.new
+----+------+---------+------+------------+------------+
| id | name | website | logo | created_at | updated_at |
+----+------+---------+------+------------+------------+
|    |      |         |      |            |            |
+----+------+---------+------+------------+------------+
1 row in set
[13] pry(main)> s.logo_changed?
false
[14] pry(main)> s.logo = nil
nil
[15] pry(main)> s.logo_changed?
true
[16] pry(main)> s.changes
{
    "logo" => [
        [0] #<LogoUploader:0x007fcc972db7a0 @model=#<Source id: nil, name: nil, website: nil, logo: nil, created_at: nil, updated_at: nil>, @mounted_as=:logo>,
        [1] #<LogoUploader:0x007fcc95a7d460 @model=#<Source id: nil, name: nil, website: nil, logo: nil, created_at: nil, updated_at: nil>, @mounted_as=:logo>
    ]
}
```

### PR DESCRIPTION
**Pull request by eavgerinos**
...something blank to a mounter that was originally blank

This is a WIP. I'm not sure it doesn't break anything yet and want to experiment with it before merging.

Feedback is more than welcome

### CODE REVIEW

**Comment by eavgerinos** [comment]
@bensie @plribeiro3000 can you take a look at this please?

**Comment by plribeiro3000** [comment]
I believe this is a safe change. I might be missing something but its a really simple change. :+1:

**Comment by bensie** [comment]
:+1:

**Comment by rusikf** [comment]
Change looks useful, but I think this line is not 'easy readable':
```if !(new_file.blank? && send(:#{column}).blank?)```
I am not sure, but may be there is a more convenient way to validate empty/blank value ?

**Comment by eavgerinos** [comment]
@rusikf I can't think of a better way than `blank?`

**Comment by bensie** [comment]
I agree @eavgerinos this is fine as-is.

**Comment by eavgerinos** [comment]
Will merge it tomorrow after giving it a last crash-test. ^_^

File: spec/orm/activerecord_spec.rb [hunk 1]
```diff
@@ -185,29 +185,57 @@ class Event < ActiveRecord::Base; end # setup a basic AR class for testing
...
-        expect(@event.image).to be_blank
+      context "when empty string is assigned" do
+        it "does nothing when" do
+          @event.image = ''
+          expect(@event.image).to be_blank
+        end
+
+        context "and the previous value was an empty string" do
+          before do
+            @event.image = ""
```
Comment (plribeiro3000) [inline spec/orm/activerecord_spec.rb]: Perhaps you can this guy above to a before block inside the context `when empty string is assigned` and get a little dry. What do you think?

File: spec/orm/activerecord_spec.rb [hunk 2]
```diff
@@ -185,29 +185,57 @@ class Event < ActiveRecord::Base; end # setup a basic AR class for testing
...
-        expect(@event.image).to be_blank
+      context "when nil is assigned" do
+        it "does nothing" do
+          @event.image = nil
+          expect(@event.image).to be_blank
+        end
+
+        context "and the previous value was nil" do
+          before do
+            @event.image = nil
```
Comment (plribeiro3000) [inline spec/orm/activerecord_spec.rb]: Same thing here. A before block in the context on line 228 would make it more dry.

**Pull request merged**